homeassistant/components/lookin/media_player.py

```diff
@@ -0,0 +1,224 @@
+
+_TYPE_TO_DEVICE_CLASS = {"01": DEVICE_CLASS_TV, "02": DEVICE_CLASS_RECEIVER}
+
+_FUNCTION_NAME_TO_FEATURE = {
+    "power": SUPPORT_TURN_OFF,
+    "poweron": SUPPORT_TURN_ON,
+    "poweroff": SUPPORT_TURN_OFF,
+    "mute": SUPPORT_VOLUME_MUTE,
+    "volup": SUPPORT_VOLUME_STEP,
+    "chup": SUPPORT_NEXT_TRACK,
+    "chdown": SUPPORT_PREVIOUS_TRACK,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the media_player platform for lookin from a config entry."""
+    lookin_data: LookinData = hass.data[DOMAIN][config_entry.entry_id]
+    entities = []
+
+    for remote in lookin_data.devices:
+        if remote["Type"] not in _TYPE_TO_DEVICE_CLASS:
+            continue
+        uuid = remote["UUID"]
+
+        def _wrap_async_update(
+            uuid: str,
+        ) -> Callable[[], Coroutine[None, Any, Remote]]:
+            """Create a function to capture the uuid cell variable."""
+
+            async def _async_update() -> Remote:
+                return await lookin_data.lookin_protocol.get_remote(uuid)
+
+            return _async_update
+
+        coordinator = DataUpdateCoordinator(
+            hass,
+            LOGGER,
+            name=f"{config_entry.title} {uuid}",
+            update_method=_wrap_async_update(uuid),
+            update_interval=timedelta(
+                seconds=60
+            ),  # Updates are pushed (fallback is polling)
+        )
+        await coordinator.async_refresh()
+        device: Remote = coordinator.data
+
+        entities.append(
+            LookinMedia(
+                uuid=uuid,
+                device=device,
+                lookin_data=lookin_data,
+                device_class=_TYPE_TO_DEVICE_CLASS[remote["Type"]],
+                coordinator=coordinator,
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class LookinMedia(LookinPowerEntity, MediaPlayerEntity):
+    """A lookin media player."""
+
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        uuid: str,
+        device: Remote,
+        lookin_data: LookinData,
+        device_class: str,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        """Init the lookin media player."""
+        self._attr_device_class = device_class
+        self._attr_supported_features: int = 0
+        self._state: str | None = None
+        self._is_muted: bool = False
+        super().__init__(coordinator, uuid, device, lookin_data)
+        for function_name, feature in _FUNCTION_NAME_TO_FEATURE.items():
+            if function_name in self._function_names:
+                self._attr_supported_features |= feature
+        self._async_update_from_data()
+
+    @property
+    def _remote(self) -> Remote:
+        return cast(Remote, self.coordinator.data)
+
+    @property
+    def state(self) -> str | None:
+        """State of the player."""
+        return self._state
+
```

Comment:
```suggestion

```

Drop this and let the base class handle it https://github.com/home-assistant/core/blob/dev/homeassistant/components/media_player/__init__.py#L440